diff --git a/cjs/visitor.js b/cjs/visitor.js
index e67d3a276301db8359b19136e8eded6c1ce92633..d13d4a33e93c1b8067e0f9f0d64e218522f006c4 100644
--- a/cjs/visitor.js
+++ b/cjs/visitor.js
@@ -190,7 +190,7 @@ class ReactApolloVisitor extends visitor_plugin_common_1.ClientSideBaseVisitor {
         return [componentProps, component].join('\n');
     }
     _buildHooksJSDoc(node, operationName, operationType) {
-        const variableString = node.variableDefinitions.reduce((acc, item) => {
+        const variableString = (node.variableDefinitions ?? []).reduce((acc, item) => {
             const name = item.variable.name.value;
             return `${acc}\n *      ${name}: // value for '${name}'`;
         }, '');
@@ -260,6 +260,7 @@ class ReactApolloVisitor extends visitor_plugin_common_1.ClientSideBaseVisitor {
           return ${this.getApolloReactHooksIdentifier()}.useLazyQuery<${operationResultType}, ${operationVariablesTypes}>(${this.getDocumentNodeVariable(node, documentVariableName)}, options);
         }`);
             hookResults.push(`export type ${lazyOperationName}HookResult = ReturnType<typeof use${lazyOperationName}>;`);
+            if (this.config.withSuspenseQuery !== false) {
             const suspenseOperationName = this.convertName(nodeName, {
                 suffix: (0, change_case_all_1.pascalCase)('SuspenseQuery'),
                 useTypesPrefix: false,
@@ -272,6 +273,7 @@ export function use${suspenseOperationName}(baseOptions?: ${this.getApolloReactH
           return ${this.getApolloReactHooksIdentifier()}.useSuspenseQuery<${operationResultType}, ${operationVariablesTypes}>(${this.getDocumentNodeVariable(node, documentVariableName)}, options);
         }`);
             hookResults.push(`export type ${suspenseOperationName}HookResult = ReturnType<typeof use${suspenseOperationName}>;`);
+            }
         }
         return [...hookFns, ...hookResults].join('\n');
     }
@@ -317,7 +319,7 @@ export function use${suspenseOperationName}(baseOptions?: ${this.getApolloReactH
             suffix: 'MutationOptions',
             useTypesPrefix: false,
         });
-        return `export type ${mutationOptionsType} = ${this.getApolloReactCommonIdentifier()}.BaseMutationOptions<${operationResultType}, ${operationVariablesTypes}>;`;
+        return `export type ${mutationOptionsType} = ${this.getApolloReactCommonIdentifier()}.MutationHookOptions<${operationResultType}, ${operationVariablesTypes}>;`;
     }
     _buildRefetchFn(node, documentVariableName, operationType, operationVariablesTypes) {
         var _a, _b;
diff --git a/esm/visitor.js b/esm/visitor.js
index 3c6d422760f567a6666659e1de88c787255c773d..a74ff046f703ddb3eb00ec61d163471ab3e97f39 100644
--- a/esm/visitor.js
+++ b/esm/visitor.js
@@ -18,6 +18,7 @@ export class ReactApolloVisitor extends ClientSideBaseVisitor {
             withMutationFn: getConfigValue(rawConfig.withMutationFn, true),
             withRefetchFn: getConfigValue(rawConfig.withRefetchFn, false),
             withFragmentHooks: getConfigValue(rawConfig.withFragmentHooks, false),
+            withSuspenseQuery: getConfigValue(rawConfig.withSuspenseQuery, true),
             apolloReactCommonImportFrom: getConfigValue(rawConfig.apolloReactCommonImportFrom, rawConfig.reactApolloVersion === 2
                 ? '@apollo/react-common'
                 : APOLLO_CLIENT_3_UNIFIED_PACKAGE),
@@ -186,7 +187,7 @@ export class ReactApolloVisitor extends ClientSideBaseVisitor {
         return [componentProps, component].join('\n');
     }
     _buildHooksJSDoc(node, operationName, operationType) {
-        const variableString = node.variableDefinitions.reduce((acc, item) => {
+        const variableString = (node.variableDefinitions ?? []).reduce((acc, item) => {
             const name = item.variable.name.value;
             return `${acc}\n *      ${name}: // value for '${name}'`;
         }, '');
@@ -256,6 +257,7 @@ export class ReactApolloVisitor extends ClientSideBaseVisitor {
           return ${this.getApolloReactHooksIdentifier()}.useLazyQuery<${operationResultType}, ${operationVariablesTypes}>(${this.getDocumentNodeVariable(node, documentVariableName)}, options);
         }`);
             hookResults.push(`export type ${lazyOperationName}HookResult = ReturnType<typeof use${lazyOperationName}>;`);
+            if (this.config.withSuspenseQuery !== false) {
             const suspenseOperationName = this.convertName(nodeName, {
                 suffix: pascalCase('SuspenseQuery'),
                 useTypesPrefix: false,
@@ -268,6 +270,7 @@ export function use${suspenseOperationName}(baseOptions?: ${this.getApolloReactH
           return ${this.getApolloReactHooksIdentifier()}.useSuspenseQuery<${operationResultType}, ${operationVariablesTypes}>(${this.getDocumentNodeVariable(node, documentVariableName)}, options);
         }`);
             hookResults.push(`export type ${suspenseOperationName}HookResult = ReturnType<typeof use${suspenseOperationName}>;`);
+            }
         }
         return [...hookFns, ...hookResults].join('\n');
     }
@@ -313,7 +316,7 @@ export function use${suspenseOperationName}(baseOptions?: ${this.getApolloReactH
             suffix: 'MutationOptions',
             useTypesPrefix: false,
         });
-        return `export type ${mutationOptionsType} = ${this.getApolloReactCommonIdentifier()}.BaseMutationOptions<${operationResultType}, ${operationVariablesTypes}>;`;
+        return `export type ${mutationOptionsType} = ${this.getApolloReactCommonIdentifier()}.MutationHookOptions<${operationResultType}, ${operationVariablesTypes}>;`;
     }
     _buildRefetchFn(node, documentVariableName, operationType, operationVariablesTypes) {
         var _a, _b;
